Add docs for cakephp/cakephp#17252 (#7695)
cakephp/docs: en/appendices/4-5-migration-guide.rst

  reference additional translation domains with `plugin_name.domain`. For
  example ``__d('DebugKit.errors', 'oh no')``.

ORM
---

``EntityTrait::$_hasAllowsNull`` was added. This property allows you to

> incrementally opt-in to a breaking change present in 5.x for ``EntityTrait::has()``

Reviewer: ```suggestion
incrementally opt-in to a breaking change present in 5.x for ``EntityTrait::has()``.
```

> When set to true, this property will result in ``has()`` and related methods use

Reviewer: ```suggestion
When set to true, this property will make ``has()`` and related methods use
```

``array_key_exists`` instead of ``isset`` to decide if fields are 'defined' in an

> entity. The semantic change is that usage like::

Reviewer: ```suggestion
entity. This will affect code like::
```

    if ($user->has('name')) { 
        // More logic
    }

In 4.x this condition would **fail** if ``name`` was ``null``. However, in 5.0,
this will condition will now **pass**. You can prepare your application for this

> change by incrementally setting ``$_hasAllowsNull``.

Author: If this is redundant let me know. I was trying to end the paragraph with something actionable. :shrug:

Reviewer: This part looks good
